Source/Urho3D/CSharp/Swig/RefCounted.i

```diff
@@ -240,7 +240,7 @@
     %apply Urho3D::SharedPtr<TYPE, BASE> { Urho3D::SharedPtr<TYPE, BASE>& }
 
     %typemap(in) Urho3D::SharedPtr<TYPE, BASE> & %{
-        $*1_ltype $1Ref($input);
+        $*1_ltype $1Ref(dynamic_cast<TYPE*>($input), $input);
```

Comment:
Whats the story behind this `dynamic_cast<>`?

Reply:
It is required to pass AbstractFile as argument.